### PR TITLE
[expo-task-manager]<feat>: Add isAvailableAsync

### LINKED ISSUE #9659: [expo-task-manager] Create isAvailableAsync method
It's important that each module has a method for getting native availability.

### PR DESCRIPTION
**Pull request by pranshuchittora**
### Why

Closes #9659

### How

Added isAvailableAsync
### Test Plan
NA - WIP

### CODE REVIEW

**Comment by pranshuchittora** [comment]
Need some help
- for iOS module
- Exported the method only for the web, is there be any export from the root file i.e. `TaskManager.ts`

File: packages/expo-task-manager/src/ExpoTaskManager.web.ts [hunk 1]
```diff
@@ -7,4 +7,7 @@ export default {
   },
   addListener() {},
   removeListeners() {},
+  async isAvailableAsync(): Promise<boolean> {
+    return true;
```
Comment (sjchmiela) [inline packages/expo-task-manager/src/ExpoTaskManager.web.ts:11]:
```suggestion
    return false;
```
I don't think task manager is available on web. 😃

File: packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerModule.java [hunk 2]
```diff
@@ -41,6 +41,11 @@ public void onCreate(ModuleRegistry moduleRegistry) {
   }
 
   //region Expo methods
+  
+  @ExpoMethod
+  public void isAvailableAsync(final Promise promise) {
+    promise.resolve(true);
```
Comment (sjchmiela) [inline packages/expo-task-manager/android/src/main/java/expo/modules/taskManager/TaskManagerModule.java:47]:
```suggestion
    promise.resolve(mTaskService != null);
```

File: packages/expo-task-manager/ios/EXTaskManager/EXTaskManager.m [hunk 3]
```diff
@@ -89,6 +89,13 @@ - (void)stopObserving {}
 
 # pragma mark - Exported methods
 
+UM_EXPORT_METHOD_AS(isAvailableAsync,
+                    isAvailable:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  resolve(@([_taskService ]));
```
Comment (sjchmiela) [inline packages/expo-task-manager/ios/EXTaskManager/EXTaskManager.m:96]:
```suggestion
  resolve(@(_taskService != nil));
```

**Review by sjchmiela — COMMENTED** [review]
Let's use one or the other way of checking if task manager is available. 👍

**Comment by sjchmiela** [comment]
Don't forget to update the CHANGELOG too and export the method!

**Comment by pranshuchittora** [comment]
r

> Don't forget to update the CHANGELOG too and export the method!

You mean exporting the method from `TaskManager.ts` as well ?

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,8 @@ Package-specific changes not released in any SDK will be added here just before
 
 ### 🎉 New features
 
+- Add isAvailableAsync to expo-task-manager ([#10657](https://github.com/expo/expo/pull/10657) by [@PranshuChittora](https://github.com/pranshuchittora))
+
```
Comment (bbarthec) [inline CHANGELOG.md:18]:
```suggestion
```
If there's a package-level CHANGELOG, the change description goes there. This one is filled during the release of the Expo Client 😉

File: packages/expo-task-manager/CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,8 @@
 
 ### 🎉 New features
 
+- Add isAvailableAsync to expo-task-manager ([#10657](https://github.com/expo/expo/pull/10657) by [@PranshuChittora](https://github.com/pranshuchittora))
```
Comment (bbarthec) [inline packages/expo-task-manager/CHANGELOG.md:9]:
```suggestion
- Added `isAvailableAsync` method. ([#10657](https://github.com/expo/expo/pull/10657) by [@PranshuChittora](https://github.com/pranshuchittora))
```

**Review by bbarthec — CHANGES_REQUESTED** [review]

**Review by bbarthec — CHANGES_REQUESTED** [review]
The missing thing is the docs adjustment: https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/task-manager.md

Do you ming adding this method to the docs as the first method under [this section](https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/task-manager.md#api)

**Comment by pranshuchittora** [comment]
> The missing thing is the docs adjustment: https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/task-manager.md
> 
> Do you ming adding this method to the docs as the first method under [this section](https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/task-manager.md#api)

Done :+1:

**Review by bbarthec — APPROVED** [review]

**Comment by pranshuchittora** [comment]
Thanks :)

**Pull request merged**